Sources/TuistGenerator/Linter/GraphLinter.swift

```diff
@@ -251,28 +251,24 @@ public class GraphLinter: GraphLinting {
     private func lintMismatchingConfigurations(graphTraverser: GraphTraversing) -> [LintingIssue] {
         let rootProjects = graphTraverser.rootProjects()
 
-        let knownConfigurations = rootProjects.reduce(into: Set()) {
-            $0.formUnion(Set($1.settings.configurations.keys))
-        }
-
-        let projectBuildConfigurations = graphTraverser.projects.compactMap { project in
-            (name: project.value.name, buildConfigurations: Set(project.value.settings.configurations.keys))
-        }
-
-        let mismatchingBuildConfigurations = projectBuildConfigurations.filter {
-            !knownConfigurations.isSubset(of: $0.buildConfigurations)
+        var reasons = Set<String>()
+        for project in rootProjects {
+            let rootProjectConfigurations = Set(project.settings.configurations.keys)
+            let sortedRootProjectConfigurations = rootProjectConfigurations.sorted()
+
+            for dependency in graphTraverser.allProjectTargetDependencies(path: project.path) {
```

Comment:
```suggestion
            let projects: Set<Project> = Set(graphTraverser.allProjectTargetDependencies(path: project.path).map(\.project))
            for project in projects { ... }
```

Afaict, we only need to operate on projects, it might be a good idea to rewrite this for loop a little bit to make that clearer.